spec/integration/command_macros_spec.rb

```diff
@@ -46,4 +47,34 @@
     user = repo.users.by_id(user.id).one
     expect(user.name).to eql('Jane')
   end
+
+  it 'allows to pass a block to configure command class' do
+    repo = Class.new(ROM::Repository[:users]) do
+      command(:create) do
+        use :upcase_name
+      end
+    end.new(rom)
+
+    user = repo.create(name: 'Jane')
+
+    persisted_user = repo.users.by_id(user.id).one
+    expect(persisted_user.name).to eql('JANE')
+  end
+
+  it 'allows to pass a block with view syntax' do
+    repo = Class.new(ROM::Repository[:users]) do
+      command :create
+      command update: :by_id do
+        use :upcase_name
+      end
```

Comment:
you just define a schema with custom type constructors:

``` ruby
class Users < ROM::Relation[:sql]
  module Types
    JSON = String.constructor(&:to_json)
  end

  schema do
    attribute :id, Serial
    attribute :settings, JSON
  end
end
```

^ this will be used by command input processor automatically and every time you pass input to a command it will use `JSON` type to construct the value for `:settings` attribute (which effectively calls `to_json` on it).